frontend/views/pages/GroupDashboard.vue

```diff
@@ -93,8 +94,14 @@ export default ({
       return Object.keys(this.currentGroupState.proposals).length > 0
     },
     isCloseToDistributionTime () {
-      const warningDate = addTimeToDate(new Date(this.groupSettings.distributionDate), -7 * DAYS_MILLIS)
-      return Date.now() >= new Date(warningDate).getTime()
+      const dDay = new Date(this.groupSettings.distributionDate)
+      const warningDate = addTimeToDate(dDay, -7 * DAYS_MILLIS)
+
+      // when (D-day - 7d) <= today < D-day
+      return Date.now() >= new Date(warningDate).getTime() && Date.now() < dDay.getTime()
+    },
+    bannerStorageKey () {
+      return `giHideDistributionWarning.${this.currentGroupId}`
     },
     showBanner () {
       return !this.hasIncomeDetails && this.isCloseToDistributionTime && !this.ephemeral.hideBanner
```

Comment:
This condition hasn't changed, and I think it's still broken, even with the new fix & improvement to `isCloseToDistributionTime`.

The issue being - the second the user adds their income details, the banner will never re-appear, because this is a chain of `&&` conditions, and `!this.hasIncomeDetails` will return `false`.

What we want is - even if their income details are set, we want the banner to still appear 7 days before the next distribution date. Changing the income details *within that 7 day period* (while the banner is visible) should dismiss the banner. So you need to be able to detect the change to the income details to get everything to work properly.

How do you do this?

Well, there are multiple ways this can be done:

- Inside of the modal for setting the income details, when "OK" is pressed (or whatever the confirmation button is), the modal can emit an event using `okTurtles.events/emit` that we can listen for inside of here, and then dismiss the banner using `closeBanner()`. This is probably the easiest way to do it as it doesn't involve touching any of the contract related code.
- We could also listen for messages to be processed and check if the message that was processed was an update to the income details data. This method is much more complicated and I don't recommend it because of that, but it would involve listening for the `EVENT_HANDLED` event and then checking the action and data of the `GIMessage` passed in to see if it was a message to update the income details for our user. Again, I don't recommend this method, it's too complicated, but I wanted to mention it just so that you're aware that such a thing is possible.

Reply:
@taoeffect thanks for clarification on what is the needed change.
rather than emitting an event for 'submit button being pressed on the income-detail modal', I've created an event named GROUP_PROFILE_UPDATE (and one-time listener to it) which is emitted once the income-detail update gone successful.
Do you think it's a better approach?

Reply:
@SebinSong 

> Do you think it's a better approach?

It's actually not a bad idea. You're thinking in the right direction, there are some issues with how you implemented that I'll comment in-line about elsewhere, but the general idea is sound.

The thing is, clicking "OK" in the Income Details modal isn't near the code that actually makes the update, and for that reason, emitting a notification there is a bit sketchy since the actual update to the income details might happen a few milliseconds later.

However, if such a notification is made in the contract, it needs to be done in a sideEffect. I'll explain elsewhere in more detail why this is.